### ISSUE TITLE
Guest Executable running as a DomainUser cannot use Net.Tcp Port Sharing Service due to LogonType of NetworkClearText

### ISSUE THREAD

**Issue by sindhudweep** [Azure/service-fabric-issues#222]
# Guest Executable using RunAs with DomainUser cannot use Net.Tcp Port Sharing Service due to FabricHost.exe spawning process with an impersonation LogonType of 8

##### Consider a hypothetical legacy application, *LegacyApp*, which:
  - Prints some diagnostic information to the console.
  - Attempts to host a WCF service. Once without port sharing and then once with port sharing.

##### The following behaviors occur in different hosting permutations:
  - When not using RunAs security at all, the ServiceHost opens successfully both with and without port sharing.

  - When running *LegacyApp* as console application manually as the DomainUser account, the ServiceHost opens successfully both with and without port sharing.
  - #### When running *LegacyApp* as a Service Fabric Guest Exectuable running as the DomainUser account, the ServiceHost only opens successfully when port sharing is not being utilized. 

##### Additional Information.
 - *FabricHost.exe* spawns *LegacyApp* with a LogonType of 8 (NetworkClearText) using impersonation, based on the Windows Event Viewer. ![image](https://cloud.githubusercontent.com/assets/23737/24516969/7c574814-154a-11e7-84f2-1f6ce0e454a3.png)
 - Recreating what FabricHost.exe is doing in LinqPad (see attached [ImpersonationLogonTypeLinq.zip](https://github.com/Azure/service-fabric-issues/files/883242/ImpersonationLogonTypeLinq.zip)) , using the portsharing service fails with a LogonType of 8.
 - ## Changing to a logon type of 4 (Batch) just works! 
![image](https://cloud.githubusercontent.com/assets/23737/24517051/cdcd8a0a-154a-11e7-99e4-b51fffabd064.png)
 - I have included an excerpt of the redirected console log file below.
 - I have included an excerpt of *LegacyApp* below and also attached *LegacyApp* along with a Complete Service Fabric Application solution as a zip archive. [NetTcpPortSharingGuestExecutable.zip](https://github.com/Azure/service-fabric-issues/files/883145/NetTcpPortSharingGuestExecutable.zip) .

```
Running As: service-dnchldap
SID: S-1-5-21-1547161642-796845957-1801674531-63830
Administrator ? True
Hosted. PortSharing is Off
Failed hosting. PortSharing is On. System.ServiceModel.CommunicationException: The service endpoint failed to listen on the URI 'net.tcp://newdevpoc113:7856/' because access was denied.  Verify that the current user is granted access in the appropriate allowAccounts section of SMSvcHost.exe.config. ---> System.ComponentModel.Win32Exception: Access is denied
   at System.ServiceModel.Activation.SharedMemory.Read(String name, String& content)... D:\ForMicrosoft\NetTcpPortSharingGuestExecutable\LegacyApp\Program.cs:line 26
```

```csharp
    internal class Program
    {
        internal static void Main(string[] args)
        {
            Console.WriteLine($"Running As: {Environment.UserName}");
            Console.WriteLine($"SID: {WindowsIdentity.GetCurrent().User}");
            Console.WriteLine($"Administrator ? {new WindowsPrincipal(WindowsIdentity.GetCurrent()).IsInRole(WindowsBuiltInRole.Administrator)}");
            Host();
            Host(true);
            Console.ReadLine();
        }

        private static void Host(bool usePortSharing = false)
        {
            try
            {
                var host = new ServiceHost(LegacyRpcService.Instance, new Uri($"net.tcp://{Dns.GetHostName()}:7856"));
                host.AddServiceEndpoint(typeof(ILegacyServiceContract), new NetTcpBinding{PortSharingEnabled = usePortSharing}, "/");
                host.Open();
                Console.WriteLine($"Hosted. PortSharing is {(usePortSharing ? "On" : "Off")}");
                host.Abort();
            }
            catch (CommunicationException cex)
            {
                Console.WriteLine($"Failed hosting. PortSharing is {(usePortSharing ? "On" : "Off")}. {cex}");
            }
        }
    }
```

**Comment by VipulM-MSFT** [comment]
@sindhudweep - Can you try setting the following attribute on your user object?

`<User Name="Domain" AccountType="DomainUser" PerformInteractiveLogon="true"
..
`
This should fix the issue. We use network logon as it is very lightweight but allow other options through configuration in your manifest.

If your service needs the user profile, there is other attribute `LoadUserProfile` that allows you to load full user profile.

**Comment by sindhudweep** [comment]
@VipulM-MSFT,

Adding `PerformInteractiveLogon="true"` as you suggested does work with the net.tcp port sharing service, and logs on with a logon type of "2" (Interactive). I have a few followup questions:

1. As you mentioned, interactive logon is quite slow, and NetworkClearText (8) is light weight but can't use shared files (port sharing service) which is a big problem for legacy applications. Is there any possibility of getting batch (4) logon?

2. The only "documentation" for this I can find at the moment is the `ServiceFabricServiceModel.xsd` schema. Is this the preferred source of documentation for SF at the moment?

3. There seems to be a `ManagedServiceAccount` `AccountType` in the schema. Does this / will this work with group managed service accounts as well?

Many thanks!

**Comment by sindhudweep** [comment]
I just tested this out with a group managed service account I created using the `ManagedServiceAccount` AccountType`. Even though that account type cannot use `PerformInteractiveLogon="true"`, it behaves much like the other options (NetworkService / LocalUser) and can use the port sharing service with no issues.

I retract my earlier requests, and I'm closing this issue!

TL;DR summary for anyone else who encounters this sort of scenario:

1. Group Managed Service Accounts (gMSA)s work using the ManagedServiceAccount account type.
2. gMSAs can use the net.tcp port sharing service without the `PerformInteractiveLogon="true"` setting.
